builder/slide_and_fold_storage.cc

```diff
@@ -224,13 +258,102 @@ class slide_and_fold : public node_mutator {
     set_result(allocate::make(op->sym, op->storage, op->elem_size, std::move(dims), body));
   }
 
+  void try_to_fold(const var& output, bool did_overlapped_fold) {
```

Comment:
`slide_and_fold`?

Reply:
This is the name of the class, so the compiler thinks I'm trying to define a constructor, so I named slide_and_fold_buffer instead